DataOrganizer/DataOrganizer.java

```diff
@@ -30,12 +30,82 @@ public void run(){
+                            continue;
+                        }
+                        if (markerBytes.get(key) == bufforByte){
+                            System.out.println("detected:" + key);
+                            if(key == "DQT")
+                            {
+                                //Pierwsze dwa elementy opisują rozmiar metadaty
+                                int[] test = new int[2];
+                                for(int i=0;i<2;i++)
+                                {
+                                    test[i] = inputStream.read();
+                                }
+                                int l=Integer.parseInt(Integer.toHexString(test[0])+Integer.toHexString(test[1]),16);
+                                //-2 bo wczśnie wyciągliśmy dwa bity z inputstream
+                                inputStream.skip(l-2);
+                            }
+                            if(key == "DHT")
+                            {
+                                int[] test = new int[2];
+                                for(int i=0;i<2;i++)
+                                {
+                                    test[i] = inputStream.read();
+                                }
+                                int l=Integer.parseInt(Integer.toHexString(test[0])+Integer.toHexString(test[1]),16);
+                                inputStream.skip(l-2);
+                            }
+                            if(key == "DRI")
+                            {
+                                int[] test = new int[2];
+                                for(int i=0;i<2;i++)
+                                {
+                                    test[i] = inputStream.read();
+                                }
+                                int l=Integer.parseInt(Integer.toHexString(test[0])+Integer.toHexString(test[1]),16);
+                                inputStream.skip(l-2);
+                            }
+                            if(key == "SOF0")
+                            {
+                                int[] test = new int[2];
+                                for(int i=0;i<2;i++)
+                                {
+                                    test[i] = inputStream.read();
+                                }
+                                int l=Integer.parseInt(Integer.toHexString(test[0])+Integer.toHexString(test[1]),16);
+                                inputStream.skip(l-2);
+                            }
+                            if(key == "SOS")
+                            {
+                                int[] test = new int[2];
+                                for(int i=0;i<2;i++)
+                                {
+                                    test[i] = inputStream.read();
+                                }
+                                int l=Integer.parseInt(Integer.toHexString(test[0])+Integer.toHexString(test[1]),16);
+                                inputStream.skip(l-2);
```

Comment:
cała funkcja zdecydowanie za długa. Powinna mieć max 10 linii. Wtedy kod się lepiej czyta. Najlepszy wyznacznik jakości kodu to ilość zapytań "co się tu kurde dzieje" na minutę. Im lepiej napisany tym lepiej się pracuje w zespole i naprawia błędy. Teraz mozesz ogarniać co tu się dzieje ale za miesiąc lub 2 już nie koniecznie. A najgorsze jak muszisz spędzić 2 godziny na wróżeniu z fusów tylko po to by ogarnąć jak działa kod.